corehq/motech/value_source.py

```diff
@@ -419,17 +419,21 @@ class CasePropertyConstantValue(ConstantValue, CaseProperty):
     pass
 
 
-def as_value_source(data: dict) -> ValueSource:
+def as_value_source(
+    value_source_config: Union[dict, JsonDict],
+) -> ValueSource:
+    if isinstance(value_source_config, JsonDict):
```

Comment:
nit: I realize that if the type annotations are correct, then the suggested code makes no difference. Does it make sense to make it a bit more flexible for passing other non-dict types (other than `JsonDict`)?
```suggestion
    if not isinstance(value_source_config, dict):
```

Reply:
Yes, I would have preferred to do that, but we can't, because `JsonDict` extends `dict`.

The **real** problem, the reason we can't just pass a `JsonDict` to `Schema.validate()`, is because `validate()` instantiates its return value with [`new = type(data)()`](https://github.com/keleshev/schema/blob/master/schema.py#L358) ... and [`type_config` is a required parameter of `JsonDict.__init__()`](https://github.com/dimagi/jsonobject/blob/master/jsonobject/containers.pyx#L126).